openwisp_users/base/models.py

```diff
@@ -224,6 +224,15 @@ class BaseOrganizationUser(models.Model):
     class Meta:
         abstract = True
 
+    def clean(self):
+        if self.user.is_owner(str(self.organization_id)) and not self.is_admin:
```

Comment:
`str(self.organization_id)` is just to ensure that if `self.organization_id` is `None` [obj.pk](https://github.com/openwisp/openwisp-users/blob/master/openwisp_users/base/models.py#L87) should not raise a `NoneType does not have pk` error

Reply:
This is not needed, see here: https://github.com/openwisp/openwisp-users/commit/85dfe0c1dc7c61926cb1e3b0888ecadeb22b4398#diff-4e7304732af6cee239b0aeb34f221e7eR81-R98
Just rebase on the current master and it will be fine.

Reply:
@nemesisdesign I rebased on master before being able to do `self.user.is_owner(str(self.organization_id))`. remember the old `is_owner` only accepted an organization instance but you see here, i am sending the `organization_id` instead.

But, because of [this test](https://github.com/openwisp/openwisp-users/blob/master/openwisp_users/tests/test_admin.py#L642), the `clean` method is called with `self.organization_id=None` since no organization is provided (which is correct as that is the objective of the test). when this happens, if i use `user.is_owner(self.organization_id)` `None` will be passed down and since `None` is neither a `UUID` nor `str` instance, [this line](https://github.com/openwisp/openwisp-users/commit/85dfe0c1dc7c61926cb1e3b0888ecadeb22b4398#diff-4e7304732af6cee239b0aeb34f221e7eR84) fails and control moves on to [this line](https://github.com/openwisp/openwisp-users/commit/85dfe0c1dc7c61926cb1e3b0888ecadeb22b4398#diff-4e7304732af6cee239b0aeb34f221e7eR87) and will try to do `pk=None.pk` which will raise the error `NoneType does not have the attribute pk`. 

I did  `user.is_owner(str(self.organization_id))`, because if `self.organization_id=None`, the string version (`"None"`) is passed down and no error will be encountered and the correct value will be returned.